### ISSUE TITLE
Huge headers cause indefinite blocking

### ISSUE THREAD

**Issue by martinklepsch** [ztellman/aleph#239]
The API I'm talking to uses Liberator in dev which adds headers for each step in it's predefined flow. These headers can contain arbitrary system data and so it happens that they get very big.

Consuming this API via `http/get` just resulted in the `deref` blocking for (as far as I can tell) forever.

I couldn't find the exact number what is too long but it seems to be somewhere between 8k & 9k. It seems that this is also about all headers combined and not just a single header's length.

If you have Boot installed you can type this into `boot repl` one by one:

``` clojure
(boot.core/set-env! :dependencies '[[aleph "0.4.2-alpha1"]])

(require '[aleph.http :as http]
         '[manifold.deferred :as d])

(defn test [length]
  (let [port    5555
        handler (fn [request]
                  (let [k "X-Long-Header"]
                    {:status 200
                     :headers {k (apply str (repeat length "b"))}
                     :body "Hello World"}))]
    (def server (http/start-server handler {:port port}))
    (-> (deref (http/get (str "http://localhost:" port)) 2000 ::timeout)
        (d/chain prn (fn [_] (prn 'shutting-down-server) (.close server))))))

(comment 
  (test 8087)

  (.close server))
```

**Comment by ztellman** [comment]
I suspect this is due to this default value: https://github.com/ztellman/aleph/blob/master/src/aleph/http/server.clj#L379.  However, the expected behavior there is either a 408 response or the connection closing, not indefinite blocking.  I'll look into it.  In the meantime, if this came up in a real-world use case, you should increase that value.

**Comment by loomis** [comment]
We have a service using Aleph (v0.4.3) that takes a filter parameter that can be arbitrarily large.  We've run into this issue of hanging connections when the parameter exceeds around 4000 characters. We've also had an issue with invalid URLs like "http://localhost:8080?alpha= 123" (space before the number) also causing the connection to hang.  We've added checks to our own client to avoid these cases, but for users hitting the API directly, it would be better if these cases returned 414/400 error codes.

**Comment by ztellman** [comment]
Apologies for not fixing this in the last release, I'll see what I can do in the next few days.

**Comment by ztellman** [comment]
This is fixed in 0.4.4-alpha1.
